packages/rpc-core/src/rpc-methods/simulateTransaction.ts

```diff
@@ -0,0 +1,160 @@
+import { Base58EncodedAddress } from '@solana/addresses';
+import { Base64EncodedWireTransaction } from '@solana/transactions';
+
+import { TransactionError } from '../transaction-error';
+import {
+    AccountInfoBase,
+    AccountInfoWithBase64EncodedData,
+    AccountInfoWithBase64EncodedZStdCompressedData,
+    AccountInfoWithJsonData,
+    Base58EncodedBytes,
+    Base64EncodedDataResponse,
+    Commitment,
+    Slot,
+    U64UnsafeBeyond2Pow53Minus1,
+} from './common';
+
+type SimulateTransactionConfigBase = Readonly<{
+    /**
+     * Commitment level to simulate the transaction at
+     * @default finalized
+     * */
+    commitment?: Commitment;
+    /** The minimum slot that the request can be evaluated at */
+    minContextSlot?: Slot;
+}>;
+
+// Both are optional booleans, but conflict - so cannot both be true
+type SigVerifyAndReplaceRecentBlockhashConfig =
+    | Readonly<{
+          /** if `true` the transaction signatures will be verified (conflicts with `replaceRecentBlockhash`) */
+          sigVerify: true;
+          /** if `true` the transaction recent blockhash will be replaced with the most recent blockhash. (conflicts with `sigVerify`) */
+          replaceRecentBlockhash?: false;
+      }>
```

Comment:
I think since these two fields directly conflict with each other, it's better to provide one or the other, but not both, at a type level.

It doesn't look like these configs affect the return type, so you don't need any generic `<T>` handling, and instead you could use "or" `|` to make sure the configs are mutually exclusive at the typecheck level. [Here's an example](https://www.typescriptlang.org/play?#code/C4TwDgpgBAyglgWwK4BsCGwIBUBOaB2AzmgMbBwD2+AwlQGZwDmAQmodALxQBKEaAJlRQgAPAG8AUFGlQA9ACp5UmVHlRaCBHGAII+YFBQQAbhBRRgFKIUSoM0YAAsHeIqXJUoGZTLUABfgg6NFQDBnw0FDgALwh+H2k1eVkEkgpNbV19AH4ALmtgHDh8RgBuOVkoAHktYEx+KDoKHCgAIxwTbRAEhTUsZygtfFsEaxQKAycMCwGOgEckCEIDEgI26BNIpHsG6eSEodp9CAAPYBhx4DyofCQEVogcctlKmu16xua2juMuiQBfAB8pQkElAkFgTAAao84HQQEcGIwoFxeAIhKJJCpelA4VAAAaFRb4mYuAjEMiUfDWJgRYBIDqEKAAdzgKHMDygpiKDDiUAAFGl8HQomQmaynASOmB0CQILw5fpmOMSABrRxsRz4gCUqhSKhsjBhPJA+SJEBBQNB4OgvBlpHlEEVwGVFDVGsIjkRTBRPD4gnwwnEPUUuLoBPNJKcZLclM8HWdbRV6s1LLZHOg0tlfIljlJgwoyygCb0BlayY9jgAdAKhSK4GK05L8YbjXCQDq9Qksw6FaXXe7Nddgih2JbAaCJHQkPg49SbMh0JhcOT3FT+QlWmwIAA2AAsAFFZxRAvwAOpwDor2MefD5ZZFEoAGgq1VqHyaLXanVAqXoTHyeBF3sa8KVvb0WG3KAADIoDEKA9DSfhikYfIAHIt3Yfc0Kgf4YIFeAjVheEIKgAAfP17TlPslQrTUIO1CRdTEPDXzeOo+U-b4fxAIA). What do you think?

Reply:
I think it's confusing as a user that they _can_ provide both configs, but won't really know that one overrides the other until they observe behavior or read the docs

Reply:
I'm moreso suggesting if you provide both as `true`, do you get a type error? Or do you have to wait for an RPC error? I think if it's the latter, we should make it the former.

However, seeing your other comment I think this is what I was getting after. As long as you get a type error for providing both I'm good with it!

Reply:
Also I just realized I fat-fingered my typescript playground and it doesn't even throw a type error lol